includes/Database_Upgrader.php

```diff
@@ -274,6 +275,59 @@ protected function rewrite_flush() {
 		}
 	}
 
+	/**
+	 * Re-index stories in Yoast SEO if permalinks are outdated.
+	 *
+	 * @return void
+	 */
+	protected function yoast_reindex_stories() {
+		if (
+			! function_exists( 'YoastSEO' ) ||
+			! class_exists( '\Yoast\WP\SEO\Repositories\Indexable_Repository' ) ||
+			! class_exists( '\Yoast\WP\SEO\Builders\Indexable_Builder' ) ||
+			! is_a( YoastSEO(), '\Yoast\WP\SEO\Main' )
+		) {
+			return;
+		}
+
+		/**
+		 * Indexable Repository.
+		 *
+		 * @var \Yoast\WP\SEO\Repositories\Indexable_Repository $repository Indexable Repository.
+		 */
+		$repository = YoastSEO()->classes->get( 'Yoast\WP\SEO\Repositories\Indexable_Repository' );
+
+		/**
+		 * Indexable Builder.
+		 *
+		 * @var \Yoast\WP\SEO\Builders\Indexable_Builder $builder Indexable Builder.
+		 */
+		$builder = YoastSEO()->classes->get( 'Yoast\WP\SEO\Builders\Indexable_Builder' );
+
+		$indexable_before = $repository->find_for_post_type_archive( Story_Post_Type::POST_TYPE_SLUG, false );
+
+		if ( ! $indexable_before || false === strpos( $indexable_before->permalink, '/web-stories//' ) ) {
+			$builder->build_for_post_type_archive( Story_Post_Type::POST_TYPE_SLUG, $indexable_before );
+		}
+
+		$all_stories = get_posts(
+			[
+				'fields'           => 'ids',
+				'suppress_filters' => false,
+				'post_type'        => [ Story_Post_Type::POST_TYPE_SLUG ],
+				'posts_per_page'   => - 1,
```

Comment:
Platform like VIP go, limit requests to 100, even if you request -1. See this sniff. 
https://github.com/Automattic/VIP-Coding-Standards/blob/bfc7fc56e18bec55a3830de06d1d3be0443585dd/WordPress-VIP-Go/ruleset.xml#L142-L144

Reply:
Limiting this query is basically what @herregroen said. Maybe a while loop of 100 chunk. I would be EXTREMELY surprised if there are sites with more than 100 stories in the wild, but you never know.

Reply:
I would be EXTREMELY surprised as well, that's why I went with this. I can change it to 100 but I don't think we need pagination / a loop.